components/nexus-security/src/main/java/org/apache/shiro/nexus/ShiroJCacheManagerAdapter.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Sonatype Nexus (TM) Open Source Version
+ * Copyright (c) 2008-present Sonatype, Inc.
+ * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
+ *
+ * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
+ * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
+ *
+ * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
+ * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
+ * Eclipse Foundation. All other trademarks are the property of their respective owners.
+ */
+package org.apache.shiro.nexus;
+
+import javax.cache.configuration.MutableConfiguration;
+import javax.cache.expiry.EternalExpiryPolicy;
+
+import org.sonatype.sisu.goodies.common.ComponentSupport;
+
+import org.apache.shiro.ShiroException;
+import org.apache.shiro.cache.Cache;
+import org.apache.shiro.cache.CacheException;
+import org.apache.shiro.cache.CacheManager;
+import org.apache.shiro.util.Destroyable;
+import org.apache.shiro.util.Initializable;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * Shiro {@link javax.cache.CacheManager} to {@link CacheManager} adapter.
+ *
+ * @since 3.0
+ */
+public class ShiroJCacheManagerAdapter
+  extends ComponentSupport
+  implements CacheManager, Initializable, Destroyable
+{
+  private final javax.cache.CacheManager cacheManager;
+
+  public ShiroJCacheManagerAdapter(final javax.cache.CacheManager cacheManager) {
+    this.cacheManager = checkNotNull(cacheManager);
+  }
+
+  @Override
+  public <K, V> Cache<K, V> getCache(final String name) throws CacheException {
+    log.debug("Getting cache: {}", name);
+
+    return new ShiroJCacheAdapter<>(this.<K,V>maybeCreateCache(name));
+  }
+
+  @Override
+  public void destroy() throws Exception {
+    cacheManager.close();
+  }
+
+  @Override
+  public void init() throws ShiroException {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove the declaration of thrown exception 'org.apache.shiro.ShiroException' which is a runtime exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)